File: LC/library/fragment-stack-manager/src/main/java/com/tieudieu/fragmentstackmanager/BaseFragmentStack.java
package com.tieudieu.fragmentstackmanager;

import android.content.Context;

/**
 * Created by chienchieu on 27/01/2016.
 */
public abstract class BaseFragmentStack extends BaseFragment {
    private String title;
    private ScreenManager mScreenManager;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof ScreenManager) {
            mScreenManager = (ScreenManager) context;
        } else {
            throw new IllegalArgumentException();
        }
    }

    protected ScreenManager getScreenManager() {
        return mScreenManager;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getFragmentTitle() {
        return title;
    }

    public boolean showBackButton() {
        return false;
    }

    public int getIndexTag() {
        return 0;
    }

}

File: LC/library/fragment-stack-manager/src/main/java/com/tieudieu/fragmentstackmanager/BaseFragmentStack.kt
package com.tieudieu.fragmentstackmanager

import android.content.Context

/**
 * Created by chienchieu on 27/01/2016.
 */
abstract class BaseFragmentStack : BaseFragment() {
    var title: String? = null
    var back = false
    protected var screenManager: ScreenManager? = null
        private set

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is ScreenManager) {
            screenManager = context as ScreenManager?
        } else {
            throw IllegalArgumentException()
        }
    }

    fun getFragmentTitle(): String? {
        return title
    }

    fun showBackButton(): Boolean {
        return back
    }

    val indexTag: Int
        get() = 0

}
